assignment-client/src/audio/AudioMixerSlavePool.h

```diff
@@ -63,49 +99,34 @@ class AudioMixerSlavePool {
         : _workerSharedData(sharedData) { setNumThreads(numThreads); }
     ~AudioMixerSlavePool() { resize(0); }
 
-    // process packets on slave threads
+    // process packets on worker threads
     void processPackets(ConstIter begin, ConstIter end);
 
-    // mix on slave threads
+    // mix on worker threads
     void mix(ConstIter begin, ConstIter end, unsigned int frame, int numToRetain);
 
-    // iterate over all slaves
+    // iterate over all workers
     void each(std::function<void(AudioMixerSlave& slave)> functor);
 
 #ifdef DEBUG_EVENT_QUEUE
     void queueStats(QJsonObject& stats);
 #endif
 
     void setNumThreads(int numThreads);
-    int numThreads() { return _numThreads; }
+    int numThreads() { return _data.numThreads; }
 
 private:
     void run(ConstIter begin, ConstIter end);
     void resize(int numThreads);
 
-    std::vector<std::unique_ptr<AudioMixerSlaveThread>> _slaves;
-
-    friend void AudioMixerSlaveThread::wait();
-    friend void AudioMixerSlaveThread::notify(bool stopping);
-    friend bool AudioMixerSlaveThread::try_pop(SharedNodePointer& node);
-
-    // synchronization state
-    Mutex _mutex;
-    ConditionVariable _slaveCondition;
-    ConditionVariable _poolCondition;
-    void (AudioMixerSlave::*_function)(const SharedNodePointer& node);
-    std::function<void(AudioMixerSlave&)> _configure;
-    int _numThreads { 0 };
-    int _numStarted { 0 }; // guarded by _mutex
-    int _numFinished { 0 }; // guarded by _mutex
-    int _numStopped { 0 }; // guarded by _mutex
+    std::vector<std::unique_ptr<AudioMixerWorkerThread>> _workers;
 
     // frame state
-    Queue _queue;
     ConstIter _begin;
     ConstIter _end;
 
     AudioMixerSlave::SharedData& _workerSharedData;
+    AudioMixerWorkerPoolData _data;
 };
 
 #endif // hifi_AudioMixerSlavePool_h
```

Comment:
worker

Reply:
We may want to name this the same thing that we name the value above so both classes use the same value in accessing the data.

Reply:
sorry I meant the comment here, looks like there’s a few comments that still use “slave” instead of “worker”.  but I understand wanting to keep the changes local to this file so not changing everywhere right now, we can do it in pieces later

Reply:
It involves a file renaming if we wanna go all the way on that, plus renaming AudioMixerSlave

Reply:
let’s save it for a larger pass, I’ve been meaning to do this for a while, I appreciate your making the changes here